wiring/src/spark_wiring_ble.cpp

```diff
@@ -2378,6 +2378,14 @@ class BleScanDelegator {
      */
     static void onScanResultCallback(const hal_ble_scan_result_evt_t* event, void* context) {
         BleScanDelegator* delegator = static_cast<BleScanDelegator*>(context);
+
+        if (!delegator->filter_.allowDuplicates()) {
+            if (delegator->isCachedDevice(event->peer_addr)) {
+                return;
+            }
+            delegator->cachedDevices_.append(event->peer_addr);
```

Comment:
@XuGuohui Do you think it would make sense to put a hard limit here on a maximum number of results?

Reply:
I think it wouldn't make sense, cos when the vector reaches the limit the device will report a large amount of duplicates, which more looks like a bug and might make the user panic. In whichever way we go we should document the limitation.

Reply:
I'm talking specifically about the case of non-duplicates. E.g. a moving device with a long scan duration.

Reply:
What about introducing an option `BleScanFilter.cachedDeviceSize()` so that user can explicitly notice the limitation, otherwise, the cache can be uncontrollable to grow?

Reply:
Doubt anyone would use it :) Some relatively large default (e.g. 512-1024) is probably better, but honestly up to you to leave as-is potentially as well.

Reply:
Cool. I'll leave it as-is, as you said it's highly unlikely happen and we can react on it when there is customers encounter a shortage of RAM.